ci/codebuild/build.sh

```diff
@@ -6,6 +6,6 @@ set -euo pipefail
 cd $CODEBUILD_SRC_DIR
 mkdir build
 cd build
-cmake .. -GNinja -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/install $@
+cmake .. -GNinja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/install $@
```

Comment:
Building the project as a shared library is important here, because it brings out any potential linking errors. If we build as a static library instead, linking errors would show up later when the final executable is built. So, for CI we definitely need to be building as a shared library.

Reply:
I changed the commit to expose `BUILD_SHARED_LIBS` as a configurable option.